data/moves.js

```diff
@@ -12014,15 +12014,13 @@ exports.BattleMovedex = {
 					return null;
 				}
 			},
-			onTryConfusion: function (target, source, effect) {
-				if (source && target !== source && effect && (!effect.infiltrates || target.side === source.side)) {
+			onTryAddVolatile: function (status, target, source, effect) {
+				if (status.id === 'confusion' && source && target !== source && effect && (!effect.infiltrates || target.side === source.side)) {
```

Comment:
Might as well combine this with the block below with `(status.id === 'confusion' || status.id === 'yawn')` instead of having the same checks twice. The debug messages aren't really needed anyway since Safeguard activates now.